Sources/AppAuthCore/OIDServiceConfiguration.m

```diff
@@ -185,8 +185,17 @@ - (nullable instancetype)initWithCoder:(NSCoder *)aDecoder {
     return nil;
   }
 
-  OIDServiceDiscovery *discoveryDocument = [aDecoder decodeObjectOfClass:[OIDServiceDiscovery class]
-                                                                  forKey:kDiscoveryDocumentKey];
+  NSSet<Class> *allowedClasses = [NSSet setWithArray:@[[OIDServiceDiscovery class],
+                                                       // The following classes are required in
+                                                       // order to support secure decoding of the
+                                                       // old OIDServiceDiscovery encoding.
+                                                       [NSDictionary class],
+                                                       [NSArray class],
+                                                       [NSString class],
+                                                       [NSNumber class],
+                                                       [NSNull class]]];
+  OIDServiceDiscovery *discoveryDocument = [aDecoder decodeObjectOfClasses:allowedClasses
+                                                                    forKey:kDiscoveryDocumentKey];
```

Comment:
if there are any unusual bug reports from users, this internal change might be something to double-check